src/middlewared/middlewared/plugins/tunables.py

```diff
@@ -25,212 +26,110 @@ class TunableService(CRUDService):
     class Config:
         datastore = 'system.tunable'
         datastore_prefix = 'tun_'
-        datastore_extend = 'tunable.upper'
         cli_namespace = 'system.tunable'
 
-    def __init__(self, *args, **kwargs):
-        super(TunableService, self).__init__(*args, **kwargs)
-        self.__default_sysctl = {}
+    SYSTEM_TUNABLES = deque()
 
     ENTRY = Patch(
         'tunable_create', 'tunable_entry',
         ('add', Int('id')),
     )
 
     @private
-    async def default_sysctl_config(self):
-        return self.__default_sysctl
+    def get_system_tunables(self):
+        if not TunableService.SYSTEM_TUNABLES:
+            tunables = subprocess.run(['sysctl', '-aN'], stdout=subprocess.PIPE)
+            for tunable in filter(lambda x: x, tunables.stdout.decode().split('\n')):
+                TunableService.SYSTEM_TUNABLES.append(tunable)
+        return TunableService.SYSTEM_TUNABLES
 
     @private
-    async def get_default_value(self, oid):
-        return self.__default_sysctl[oid]
+    def get_original_value(self, var):
+        with open(f'/proc/sys/{var.replace(".", "/")}') as f:
+            return f.read().strip()
 
     @private
-    async def set_default_value(self, oid, value):
-        if oid not in self.__default_sysctl:
-            self.__default_sysctl[oid] = value
+    def set_original_value(self, var, orig_value):
+        with open(f'/proc/sys/{var.replace(".", "/")}', 'w') as f:
+            f.write(orig_value)
 
     @accepts()
-    @returns(Dict(
-        'tunable_type_choices',
-        *[Str(k, enum=[k]) for k in TUNABLE_TYPES],
-    ))
+    @returns(Dict('tunable_type_choices', *[Str(k, enum=[k]) for k in TUNABLE_TYPES]))
     async def tunable_type_choices(self):
         """
-        Retrieve tunable type choices supported in the system
+        Retrieve the supported tunable types that can be changed.
         """
         return {k: k for k in TUNABLE_TYPES}
 
     @accepts(Dict(
         'tunable_create',
-        Str('var', validators=[Match(r'^[\w\.\-]+$')], required=True),
+        Str('var', required=True),
         Str('value', required=True),
-        Str('type', enum=TUNABLE_TYPES, required=True),
+        Str('type', enum=TUNABLE_TYPES, default='SYSCTL', required=True),
```

Comment:
Is this required anymore now that there's only one type of tunable ?

Reply:
Nope, definitely not